source/common/router/router.cc

```diff
@@ -585,11 +585,24 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
     headers.removeEnvoyUpstreamRequestTimeoutAltResponse();
   }
 
-  include_attempt_count_ = route_entry_->includeAttemptCount();
-  if (include_attempt_count_) {
+  include_attempt_count_in_request_ = route_entry_->includeAttemptCountInRequest();
+  if (include_attempt_count_in_request_) {
     headers.setEnvoyAttemptCount(attempt_count_);
   }
 
+  // The router has reached a point where it is going to try to send a request upstream,
+  // so now modify_headers should attach x-envoy-attempt-count to the downstream response if the
+  // config flag is true.
```

Comment:
Oh, I like this as a clear insertion point!

Reply:
Yeah, I think it came out nicely :)